blackbird/src/main/java/com/fasterxml/jackson/module/blackbird/CrossLoaderAccess.java

```diff
@@ -109,24 +109,29 @@ private static Class<?> accessClassIn(MethodHandles.Lookup lookup) throws IOExce
         try {
             return Class.forName(pkg.getName() + "." + CLASS_NAME, true, lookup.lookupClass().getClassLoader());
         } catch (ClassNotFoundException ign) { }
-        String fqcn = pkg.getName()
-                .replace('.', '/')
-            + "/" + CLASS_NAME;
-        ByteArrayOutputStream classBytes = new ByteArrayOutputStream(HEADER.length + FOOTER.length + fqcn.length() + 16);
-        DataOutputStream dataOut = new DataOutputStream(classBytes);
-        for (int b : HEADER) {
-            dataOut.writeByte(b);
-        }
-        dataOut.writeUTF(fqcn);
-        for (int b : FOOTER) {
-            dataOut.writeByte(b);
-        }
-        try {
-            return (Class<?>) DEFINE_CLASS.invokeExact(lookup, classBytes.toByteArray());
-        } catch (RuntimeException | Error | IOException | ReflectiveOperationException e) {
-            throw e;
-        } catch (Throwable e) {
-            throw new RuntimeException(e);
+        synchronized(CrossLoaderAccess.class) {
+            try {
+                return Class.forName(pkg.getName() + "." + CLASS_NAME, true, lookup.lookupClass().getClassLoader());
+            } catch (ClassNotFoundException ign) { }
+            String fqcn = pkg.getName()
+                    .replace('.', '/')
+                + "/" + CLASS_NAME;
+            ByteArrayOutputStream classBytes = new ByteArrayOutputStream(HEADER.length + FOOTER.length + fqcn.length() + 16);
+            DataOutputStream dataOut = new DataOutputStream(classBytes);
+            for (int b : HEADER) {
+                dataOut.writeByte(b);
+            }
+            dataOut.writeUTF(fqcn);
+            for (int b : FOOTER) {
+                dataOut.writeByte(b);
+            }
+            try {
+                return (Class<?>) DEFINE_CLASS.invokeExact(lookup, classBytes.toByteArray());
+            } catch (RuntimeException | Error | IOException | ReflectiveOperationException e) {
+                throw e;
+            } catch (Throwable e) {
```

Comment:
Could be `Exception` although I suppose does not make much difference as preceding catch block already handles `RuntimeException` and `Error`.

Reply:
possibly.  I didn't want to mess with the existing code much so left it as is other than surrounding with synchronization....

Reply:
`invokeExact` declares `throws Throwable` and since it's possible to declare your own subtype of `Throwable`, the `RuntimeException | Error` doesn't actually cover all possible types. So I think making this change would result in a compiler error. Feel free to play around with it though and see if you can do better!

Reply:
Interesting! Did not realize that someone would mark `Throwable` to be thrown -- don't recall ever seeing that -- but if so, yes, that makes sense.
(also don't remember ever seeing custom `Throwable` sub-class but now I know :) ).

Reply:
This is because `invokeExact` represents an arbitrary method call, with unknown `throws` declared. So since you can call a method that might `throws Throwable`, so too `invokeExact` must handle this possibility.

This was not needed in classic reflection, since any thrown exception was wrapped in an `InvocationTargetException`. But, this didn't end up always being good - for example if you wrapped an unexpected `Error` often the exception handling would get very confusing.